demos/fleet_provisioning/fleet_provisioning_with_csr/fleet_provisioning_csr_demo.c

```diff
@@ -0,0 +1,874 @@
+
+        if( status == false )
+        {
+            LogError( ( "Failed to unsubscribe from fleet provisioning topic: %.*s.",
+                        FP_CBOR_CREATE_CERT_REJECTED_LENGTH,
+                        FP_CBOR_CREATE_CERT_REJECTED_TOPIC ) );
+        }
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool subscribeToRegisterThingResponseTopics( void )
+{
+    bool status;
+
+    status = SubscribeToTopic( FP_CBOR_REGISTER_ACCEPTED_TOPIC( PROVISIONING_TEMPLATE_NAME ),
+                               FP_CBOR_REGISTER_ACCEPTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ) );
+
+    if( status == false )
+    {
+        LogError( ( "Failed to subscribe to fleet provisioning topic: %.*s.",
+                    FP_CBOR_REGISTER_ACCEPTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ),
+                    FP_CBOR_REGISTER_ACCEPTED_TOPIC( PROVISIONING_TEMPLATE_NAME ) ) );
+    }
+
+    if( status == true )
+    {
+        status = SubscribeToTopic( FP_CBOR_REGISTER_REJECTED_TOPIC( PROVISIONING_TEMPLATE_NAME ),
+                                   FP_CBOR_REGISTER_REJECTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ) );
+
+        if( status == false )
+        {
+            LogError( ( "Failed to subscribe to fleet provisioning topic: %.*s.",
+                        FP_CBOR_REGISTER_REJECTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ),
+                        FP_CBOR_REGISTER_REJECTED_TOPIC( PROVISIONING_TEMPLATE_NAME ) ) );
+        }
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool unsubscribeFromRegisterThingResponseTopics( void )
+{
+    bool status;
+
+    status = UnsubscribeFromTopic( FP_CBOR_REGISTER_ACCEPTED_TOPIC( PROVISIONING_TEMPLATE_NAME ),
+                                   FP_CBOR_REGISTER_ACCEPTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ) );
+
+    if( status == false )
+    {
+        LogError( ( "Failed to unsubscribe from fleet provisioning topic: %.*s.",
+                    FP_CBOR_REGISTER_ACCEPTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ),
+                    FP_CBOR_REGISTER_ACCEPTED_TOPIC( PROVISIONING_TEMPLATE_NAME ) ) );
+    }
+
+    if( status == true )
+    {
+        status = UnsubscribeFromTopic( FP_CBOR_REGISTER_REJECTED_TOPIC( PROVISIONING_TEMPLATE_NAME ),
+                                       FP_CBOR_REGISTER_REJECTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ) );
+
+        if( status == false )
+        {
+            LogError( ( "Failed to unsubscribe from fleet provisioning topic: %.*s.",
+                        FP_CBOR_REGISTER_REJECTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ),
+                        FP_CBOR_REGISTER_REJECTED_TOPIC( PROVISIONING_TEMPLATE_NAME ) ) );
+        }
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool saveCertificate( const char * pCertificate,
```

Comment:
As certificate is in DER format, it would be important to represent it in `uint8_t *` type. 
```suggestion
static bool saveCertificate( const uint8_t * pCertificate,
```

Reply:
This change can be considered in a future PR when DER format is being used for credentials